test/hotspot/jtreg/runtime/locking/TestUnstructuredLocking.jasm

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @summary Check that monitorenter A; monitorenter B; monitorexit A; monitorexit B; works
+ * @compile TestUnstructuredLocking.jasm
+ * @run main/othervm -Xint TestUnstructuredLocking
+ */
+
+super public class TestUnstructuredLocking version 64:0 {
+
+    public static Method main:"([Ljava/lang/String;)V" stack 2 locals 4 {
+        new class java/lang/Object;
+        dup;
+        invokespecial Method java/lang/Object."<init>":"()V";
+        astore_1;
+        new class java/lang/Object;
+        dup;
+        invokespecial Method java/lang/Object."<init>":"()V";
+        astore_2;
+        aload_1;
+        monitorenter;
+        aload_2;
+        monitorenter;
+        aload_1;
+        monitorexit;
+        aload_2;
+        invokevirtual Method java/lang/Object.notify:"()V";
```

Comment:
Curious why this is needed.

Reply:
I don't think it is needed. It's probably a left-over from an experiment that I did, maybe to avoid this whole method to be optimized-away (but that shouldn't matter, because it's not JIT-compiled anyway) or more likely to see how this interacts with inflation. In any case, I'll remove it.